Riot/Modules/Common/SwiftUI/VectorHostingBottomSheetPreferences.swift

```diff
@@ -0,0 +1,118 @@
+class VectorHostingBottomSheetPreferences {
+    
+    // MARK: - Detent
+    
+    enum Detent {
+        case medium
+        case large
+        
+        @available(iOS 15, *)
+        fileprivate func uiSheetDetent() -> UISheetPresentationController.Detent {
+            switch self {
+            case .medium: return .medium()
+            case .large: return .large()
+            }
+        }
+        
+        @available(iOS 15, *)
+        fileprivate func uiSheetDetentId() -> UISheetPresentationController.Detent.Identifier {
+            switch self {
+            case .medium: return .medium
+            case .large: return .large
+            }
+        }
+    }
+    
+    // MARK: - Public
+    
+    // The array of detents that the sheet may rest at.
+    // This array must have at least one element.
+    // Detents must be specified in order from smallest to largest height.
+    // Default: [.medium, .large]
+    let detents: [Detent]
+    
+    // The default detent. When nil or the identifier is not found in detents, the sheet is displayed at the smallest detent.
+    // Default: nil
+    let defaultDetent: Detent?
+    
+    // The largest detent that is not dimmed. When nil or the identifier is not found in detents, all detents are dimmed.
+    // Default: nil
+    let largestUndimmedDetent: Detent?
+    let cornerRadius: CGFloat?
+    
+    // If there is a larger detent to expand to than the selected detent, and a descendent scroll view is scrolled to top, this controls whether scrolling down will expand to a larger detent.
+    // Useful to set to NO for non-modal sheets, where scrolling in the sheet should not expand the sheet and obscure the content above.
+    // Default: YES
+    let prefersScrollingExpandsWhenScrolledToEdge: Bool
+    
+    // Set to YES to show a grabber at the top of the sheet.
+    // Default: `nil` -> the grabber is shown if more than one detent is configured
+    let prefersGrabberVisible: Bool?
+    
+    // MARK: - Setup
+    
+    init(detents: [Detent] = [.medium, .large],
+         defaultDetent: Detent? = nil,
+         largestUndimmedDetent: Detent? = nil,
+         prefersGrabberVisible: Bool? = nil,
+         cornerRadius: CGFloat? = nil,
+         prefersScrollingExpandsWhenScrolledToEdge: Bool = true) {
+        self.detents = detents
+        self.defaultDetent = defaultDetent
+        self.largestUndimmedDetent = largestUndimmedDetent
+        self.prefersGrabberVisible = prefersGrabberVisible
+        self.cornerRadius = cornerRadius
+        self.prefersScrollingExpandsWhenScrolledToEdge = prefersScrollingExpandsWhenScrolledToEdge
+    }
+    
+    // MARK: - Public
+    
+    func setup(viewController: UIViewController) {
+        if #available(iOS 15.0, *) {
+            if let sheetController = viewController.sheetPresentationController {
```

Comment:
```suggestion
            guard let sheetController = viewController.sheetPresentationController else {
                return 
            }
```
Or in the same guard above

Reply:
Makes sense!